### DISCUSSION TITLE
Using GitHub API in forked project

### DISCUSSION THREAD

**Discussion by valhuber** [community/community#30973] [Codespaces]
I am unclear how to use the [GitHub APIs](https://docs.github.com/en/rest/codespaces) when my project is forked.

For obvious reasons, the APIs require a {token}, which I take to be a [PAT - Personal Access Token](https://github.com/settings/tokens).

But [my project](https://github.com/valhuber/ApiLogicServer#readme) is an application builder, which I expect _other folks_ to fork and run.  When they run, I want to compute _their_ URL address.

I do so now with this:

```
    use_codespace_defaulting = True  # experimental support to run default launch config
    if use_codespace_defaulting and os.getenv('CODESPACES') and swagger_host == 'localhost':
        app_logger.info('\n Applying Codespaces default port settings')
        swagger_host = os.getenv('CODESPACE_NAME') + '-5656.githubpreview.dev'
        swagger_port = '443'
        http_type = 'https'
```

This code works to obtain the `swagger_host`, but I am advised this would be best performed using the GitHub APIs.  However, this needs to run in the _forked_ project, where my PAT is presumably invalid.

So, my question is whether this code is ill-advised, or whether there is a way to use the APIs to obtain the `swagger_host`.

**Comment by greggroth** [comment]
Your PAT will not provide you access to another users codespaces, so they would need to create their own PAT to use the API. One option is to prompt the user to generate a PAT with the `codespace` scope, set it as a [codespaces secret](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces#adding-a-secret) and update the script so it's making a request using the codespace owners PAT.

**Comment by valhuber** [comment]
HI, @greggroth 

Thanks for getting back.

The key objective I had was to make this simple, so prompting etc may not be simpler.

Codespaces provides useful ENV variables now... maybe add one for the URL?
